core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -265,30 +265,33 @@ protected[core] object WhiskWebActionsApi extends Directives {
     }
 
     /**
-     * Finds the content-type in the header list and maps it to a known media type. If it is not
-     * recognized, construct a failure with appropriate message.
+     * Finds the content-type in the header list and ensures that it is a valid format. If it is not
+     * valid, construct a failure with appropriate message.
+     * If the content-type header is missing, then return the supplied defaultType
      */
     private def findContentTypeInHeader(headers: List[RawHeader], transid: TransactionId, defaultType: MediaType): Try[MediaType] = {
         headers.find(_.lowercaseName == `Content-Type`.lowercaseName) match {
             case Some(header) =>
                 MediaType.parse(header.value) match {
-                    case Right(mediaType: MediaType) =>
-                        // lookup the media type specified in the content header to see if it is a recognized type
-                        MediaTypes.getForKey(mediaType.mainType -> mediaType.subType).map(Success(_)).getOrElse {
-                            // this is a content-type that is not recognized, reject it
-                            Failure(RejectRequest(BadRequest, Messages.httpUnknownContentType)(transid))
-                        }
+                    case Right(mediaType: MediaType) => Success(mediaType) // mediaType is a valid format
```

Comment:
@akrabat, please check to make sure this is correct even when an invalid media type is passed. If I recall correctly, there was a reason we were using `getForKey`.